scipy/signal/_peak_finding.py

```diff
@@ -376,34 +485,9 @@ def peak_prominences(x, peaks, wlen=None):
     only two candidates in the evaluated area are the two neighbouring samples
     and a smaller prominence is calculated.
     """
-    # Inner function expects `x` to be C-contiguous
-    x = np.asarray(x, order='C', dtype=np.float64)
-    if x.ndim != 1:
-        raise ValueError('`x` must have exactly one dimension')
-
-    peaks = np.asarray(peaks)
-    if peaks.size == 0:
-        # Empty arrays default to np.float64 but are valid input
-        peaks = np.array([], dtype=np.intp)
-    try:
-        # Safely convert to C-contiguous array of type np.intp
-        peaks = peaks.astype(np.intp, order='C', casting='safe',
-                             subok=False, copy=False)
-    except TypeError:
-        raise TypeError("Cannot safely cast `peaks` to dtype('intp')")
-    if peaks.ndim != 1:
-        raise ValueError('`peaks` must have exactly one dimension')
-
-    if wlen is None:
-        wlen = -1  # Inner function expects int -> None == -1
-    elif 1 < wlen:
-        # Round up to next positive integer; rounding up to next odd integer
-        # happens implicitly inside the inner function
-        wlen = int(math.ceil(wlen))
-    else:
-        # Give feedback if wlen has unexpected value
-        raise ValueError('`wlen` must be at larger than 1, was ' + str(wlen))
-
+    x = _handle_arg_x(x)
```

Comment:
My initial `_check_for_1D` argument is made for these three. There is no answer to *What is handled here?* from a readability point of view.

Reply:
I agree the naming should be clearer.